massband/rdf/core.py

```
@@ -0,0 +1,439 @@
+                n_molecules = len(mol_indices_tuples)
+                if n_molecules == 0:
+                    continue
+                n_atoms_per_mol = len(mol_indices_tuples[0])
+                atom_indices = jnp.array(mol_indices_tuples)
+                mol_masses = jnp.array(masses[name])
+
+                # Process frames sequentially with jax.lax.map
+                def process_frame(frame_args):
+                    frame_positions, frame_cell = frame_args
+                    return _calculate_pbc_aware_com_for_frame(
+                        frame_positions, frame_cell, atom_indices, mol_masses, n_molecules, n_atoms_per_mol
```

Ruff (0.12.2)

Function definition does not bind loop variable `atom_indices`

(B023)